From 26f9d69621733703998f997aba9ac89c16d6317e Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Tue, 27 Sep 2016 09:26:21 +0800
Subject: [PATCH 702/827] drm: xilinx: fix compiling errors

This patch comes from:
  https://github.com/Xilinx/linux-xlnx.git

drivers/gpu/drm/xilinx/xilinx_drm_drv.c:420:2: warning: (near initialization for 'xilinx_drm_driver.disable_vblank')
  CC      drivers/gpu/drm/xilinx/xilinx_drm_fb.o
drivers/gpu/drm/xilinx/xilinx_drm_fb.c: In function 'xilinx_drm_fb_alloc':
drivers/gpu/drm/xilinx/xilinx_drm_fb.c:106:2: warning: passing argument 2 of 'drm_helper_mode_fill_fb_struct' discards 'const' qualifier from pointer target type
  drm_helper_mode_fill_fb_struct(&fb->base, mode_cmd);
  ^
In file included from drivers/gpu/drm/xilinx/xilinx_drm_fb.c:24:0:
include/drm/drm_crtc_helper.h:199:13: note: expected 'struct drm_mode_fb_cmd2 *' but argument is of type 'const struct drm_mode_fb_cmd2 *'
 extern void drm_helper_mode_fill_fb_struct(struct drm_framebuffer *fb,
             ^
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/gpu/drm/xilinx/xilinx_drm_crtc.c | 5 +++--
 drivers/gpu/drm/xilinx/xilinx_drm_drv.c  | 4 +++-
 drivers/gpu/drm/xilinx/xilinx_drm_fb.c   | 3 ++-
 3 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/xilinx/xilinx_drm_crtc.c b/drivers/gpu/drm/xilinx/xilinx_drm_crtc.c
index 6bc4212..33b036b 100644
--- a/drivers/gpu/drm/xilinx/xilinx_drm_crtc.c
+++ b/drivers/gpu/drm/xilinx/xilinx_drm_crtc.c
@@ -225,7 +225,8 @@ static int xilinx_drm_crtc_mode_set_base(struct drm_crtc *base_crtc,
 					 struct drm_framebuffer *old_fb)
 {
 	/* configure a plane */
-	return _xilinx_drm_crtc_mode_set_base(base_crtc, base_crtc->fb, x, y);
+	return _xilinx_drm_crtc_mode_set_base(base_crtc, base_crtc->primary->fb,
+               x, y);
 }
 
 /* load rgb LUT for crtc */
@@ -335,7 +336,7 @@ static int xilinx_drm_crtc_page_flip(struct drm_crtc *base_crtc,
 		return ret;
 	}
 
-	base_crtc->fb = fb;
+	base_crtc->primary->fb = fb;
 
 	if (event) {
 		event->pipe = 0;
diff --git a/drivers/gpu/drm/xilinx/xilinx_drm_drv.c b/drivers/gpu/drm/xilinx/xilinx_drm_drv.c
index cc78583..f6af786 100644
--- a/drivers/gpu/drm/xilinx/xilinx_drm_drv.c
+++ b/drivers/gpu/drm/xilinx/xilinx_drm_drv.c
@@ -504,7 +504,9 @@ static int xilinx_drm_platform_probe(struct platform_device *pdev)
 /* exit xilinx drm platform */
 static int xilinx_drm_platform_remove(struct platform_device *pdev)
 {
-	drm_platform_exit(&xilinx_drm_driver, pdev);
+	struct xilinx_drm_private *private = platform_get_drvdata(pdev);
+
+	drm_put_dev(private->drm);
 
 	return 0;
 }
diff --git a/drivers/gpu/drm/xilinx/xilinx_drm_fb.c b/drivers/gpu/drm/xilinx/xilinx_drm_fb.c
index 3890fea..f3d7372 100644
--- a/drivers/gpu/drm/xilinx/xilinx_drm_fb.c
+++ b/drivers/gpu/drm/xilinx/xilinx_drm_fb.c
@@ -147,9 +147,10 @@ int xilinx_drm_fb_helper_pan_display(struct fb_var_screeninfo *var,
 	int ret = 0;
 	int i;
 
-	if (__drm_modeset_lock_all(dev, !!oops_in_progress))
+	if (oops_in_progress)
 		return -EBUSY;
 
+	drm_modeset_lock_all(dev);
 	for (i = 0; i < fb_helper->crtc_count; i++) {
 		modeset = &fb_helper->crtc_info[i].mode_set;
 
-- 
2.9.3

